src/render/VolumeRaycastShader.cpp

```diff
@@ -54,186 +54,183 @@ VolumeRaycastShader::VolumeRaycastShader() : m_use_blending{false}, m_blend_volu
-      "sample = texture(volume, dataPos).aaaa; \n"
-      "}else if (channel == 5){ \n"
-      "sample = texture(volume, dataPos); \n"
-      "}else{ \n"
-      "sample = texture(volume, dataPos); \n"
-      "sample.a = max(sample.r, max(sample.g,sample.b)) ; "
-      "}\n"
-
-      "if(useBlend){ \n"
-      "vec4 sample_blend; \n"
-      "if (channel == 1){ \n"
-      "sample_blend = texture(blendVolume, dataPos).rrrr; \n"
-      "}else if (channel == 2){ \n"
-      "sample_blend = texture(blendVolume, dataPos).gggg; \n"
-      "}else if (channel == 3){ \n"
-      "sample_blend = texture(blendVolume, dataPos).bbbb; \n"
-      "}else if (channel == 4){ \n"
-      "sample_blend = texture(blendVolume, dataPos).aaaa; \n"
-      "}else if (channel == 5){ \n"
-      "sample_blend = texture(blendVolume, dataPos); \n"
-      "}else{ \n"
-      "sample_blend = texture(blendVolume, dataPos); \n"
-      "sample_blend.a = max(sample_blend.r, max(sample_blend.g,sample_blend.b)) ; "
-      "}\n"
-
-      "sample = (vec4(1.0f - blendAlpha) * sample) +  (vec4(blendAlpha) * sample_blend);\n"
-      "}\n"
-
-      // threshold based on alpha
-      "sample.a = (sample.a > threshold) ? sample.a : 0.0f ;\n"
-
-      // transferfunction
-      "if(useLut) {\n"
-      "if(useMultiLut){\n"
-      "sample.r = texture(lut, vec2(sample.r,0.5)).r;"
-      "sample.g = texture(lut, vec2(sample.g,0.5)).g;"
-      "sample.b = texture(lut, vec2(sample.b,0.5)).b;"
-      "sample.a = max(sample.r, max(sample.g,sample.b)) ; "
-      "}else{\n"
-      "sample = texture(lut, vec2(sample.a,0.5));"
-      "}\n"
-      "}\n"
-
-      // assume alpha is the highest channel and gamma correction
-      "sample.a = sample.a * multiplier; \n" /// needs changing
-
-      // blending (front to back)
-      "vFragColor.rgb += (1.0 - vFragColor.a) * sample.a * sample.rgb;\n"
-      "vFragColor.a += (1.0 - vFragColor.a) * sample.a;\n"
-
-      // early exit if opacity is reached
-      "if (vFragColor.a >= 0.95) \n"
-      "break;\n"
-
-      // advance point
-      "dataPos += geomDir * dt; \n"
-      "} \n"
-
-      // remove fragments for correct depthbuffer
-      "if (vFragColor.a == 0.0f)"
-      "discard;"
-      "}\n";
+    "#version 330 core\n"
+    "vec2 intersect_box(vec3 orig, vec3 dir, vec3 clip_min, vec3 clip_max) { \n"
+    "dir += vec3(0.0000001); \n"
+    "vec3 inv_dir = 1.0 / dir; \n"
+    "vec3 tmin_tmp = (clip_min - orig) * inv_dir; \n"
+    "vec3 tmax_tmp = (clip_max - orig) * inv_dir; \n"
+    "vec3 tmin = min(tmin_tmp, tmax_tmp); \n"
+    "vec3 tmax = max(tmin_tmp, tmax_tmp); \n"
+    "float t0 = max(tmin.x, max(tmin.y, tmin.z)); \n"
+    "float t1 = min(tmax.x, min(tmax.y, tmax.z)); \n"
+    "return vec2(t0, t1); \n"
+    "}\n"
+
+    "layout(location = 0) out vec4 vFragColor; \n"	//fragment shader output
+    "smooth in vec3 vUV; \n"						//3D texture coordinates form vertex shader interpolated by rasterizer
+    "uniform sampler3D volume;\n"					//volume dataset
```

Comment:
Fix the spacing on these inline comments?

Reply:
I reformatted the file.